src/probnum/linops/_linear_operator.py

```diff
@@ -1079,3 +1114,98 @@ def _astype(
             return self
         else:
             return Identity(self.shape, dtype=dtype)
+
+    def __eq__(self, other: LinearOperator) -> bool:
+        return self._is_type_shape_dtype_equal(other)
+
+
+class Selection(LinearOperator):
+    def __init__(self, indices, shape, dtype=np.double):
+        if np.ndim(indices) > 1:
+            raise ValueError(
+                "Selection LinOp expects an integer or (1D) iterable of "
+                f"integers. Received {type(indices)} with shape {np.shape(indices)}."
+            )
+        if shape[0] > shape[1]:
+            raise ValueError(
+                f"Invalid shape {shape} for Selection LinOp. If the "
+                "output-dimension (shape[0]) is larger than the input-dimension "
+                "(shape[1]), consider using `Embedding`."
+            )
+        self._indices = probnum.utils.as_shape(indices)
+        assert len(self._indices) == shape[0]
+
+        super().__init__(
+            dtype=dtype,
+            shape=shape,
+            matmul=lambda x: _selection_matmul(self.indices, x),
+            todense=self._todense,
+            transpose=lambda: Embedding(
+                take_indices=np.arange(len(self._indices)),
+                put_indices=self._indices,
+                shape=(self.shape[1], self.shape[0]),
+            ),
+        )
+
+    @property
+    def indices(self):
+        return self._indices
+
+    def _todense(self):
+        res = np.eye(self.shape[1], self.shape[1])
+        return _selection_matmul(self.indices, res)
+
+
+def _selection_matmul(indices, M):
+    return np.take(M, indices=indices, axis=-2)
+
+
+class Embedding(LinearOperator):
```

Comment:
I think this one needs a little bit of class documentation since the name is somewhat non-obvious.
Btw. another name for a similar operation coming from the DL world would be `Scatter` which I personally find less confusing since it already indicates some sort of indexing stuff

Reply:
Out of scope for this PR; requires decision in discussion #573